packages/slider/src/useSlider.ts

```diff
@@ -0,0 +1,292 @@
+    (event: MouseEvent) => {
+      let retVal = undefined;
+
+      if (trackRect) {
+        const offset = rtl ? trackRect.left + trackRect.width : trackRect.left;
+        const mouseX = (event.pageX - offset) * (rtl ? -1 : 1);
+        const x = (((max - min) * mouseX) / trackRect.width) as unknown as string;
+        const value = min + parseInt(x, 10);
+
+        // Reduce return value to align with step size.
+        retVal = Math.floor(value / step) * step;
+      }
+
+      return retVal;
+    },
+    [max, min, trackRect, rtl, step]
+  );
+
+  const setThumbPosition = useCallback(
+    (thumb: 'min' | 'max') => (value: number) => {
+      if (!disabled) {
+        let newMinValue = position.minValue;
+        let newMaxValue = position.maxValue;
+
+        if (thumb === 'min') {
+          if (value < min) {
+            newMinValue = min;
+          } else if (value > position.maxValue) {
+            newMinValue = position.maxValue;
+          } else {
+            newMinValue = value;
+          }
+        } else if (thumb === 'max') {
+          if (value > max) {
+            newMaxValue = max;
+          } else if (value < position.minValue) {
+            newMaxValue = position.minValue;
+          } else {
+            newMaxValue = value;
+          }
+        }
+
+        if (newMinValue !== position.minValue || newMaxValue !== position.maxValue) {
+          setPosition({ minValue: newMinValue, maxValue: newMaxValue });
+        }
+      }
+    },
+    [disabled, max, min, position.maxValue, position.minValue, setPosition]
+  );
+
+  /*
+   * Prop getters
+   */
+
+  const getTrackProps = useCallback<IUseSliderReturnValue['getTrackProps']>(
+    ({ onMouseDown, ...other } = {}) => {
+      const handleMouseDown = (event: MouseEvent) => {
+        if (!disabled && event.button === 0) {
+          const value = getTrackPosition(event);
+
+          if (value !== undefined) {
+            const minOffset = Math.abs(position.minValue - value);
+            const maxOffset = Math.abs(position.maxValue - value);
+
+            if (minOffset <= maxOffset) {
+              minThumbRef.current && minThumbRef.current.focus();
+              setThumbPosition('min')(value);
+            } else {
+              maxThumbRef.current && maxThumbRef.current.focus();
+              setThumbPosition('max')(value);
+            }
+          }
+        }
+      };
+
+      return {
+        'data-garden-container-id': 'containers.slider.track',
+        'data-garden-container-version': PACKAGE_VERSION,
+        'aria-disabled': disabled,
+        onMouseDown: composeEventHandlers(onMouseDown, handleMouseDown),
+        ...other
+      };
+    },
+    [
+      disabled,
+      getTrackPosition,
+      maxThumbRef,
+      minThumbRef,
+      position.maxValue,
+      position.minValue,
+      setThumbPosition
+    ]
+  );
+
+  const getThumbProps = useCallback(
+    (thumb: 'min' | 'max'): IUseSliderReturnValue['getMinThumbProps'] =>
```

Comment:
Since this getter is used for both thumbs, it's a little confusing to see the return type reference only one of the thumbs (`getMinThumbProps`). Is there an opportunity here to create one centralized type? Or to change the naming here a little so it's representative of both thumbs?